packages/nextjs/src/app-router/client/useAwaitableNavigate.ts

```diff
@@ -1,46 +1,61 @@
 'use client';
 
-import { useRouter } from 'next/navigation';
-import { useEffect, useRef, useTransition } from 'react';
+import { usePathname, useRouter } from 'next/navigation';
+import { useCallback, useEffect } from 'react';
 
-type NavigateFunction = ReturnType<typeof useRouter>['push'];
+declare global {
+  interface Window {
+    __internal_clerk_navPromisesBuffer: Array<() => void> | undefined;
+    __internal_clerk_navFun: (to: string) => Promise<void>;
+  }
+}
 
 /**
  * Creates an "awaitable" navigation function that will do its best effort to wait for Next.js to finish its route transition.
- *
  * This is accomplished by wrapping the call to `router.push` in `startTransition()`, which should rely on React to coordinate the pending state. We key off of
  * `isPending` to flush the stored promises and ensure the navigates "resolve".
  */
 export const useAwaitableNavigate = () => {
   // eslint-disable-next-line @typescript-eslint/unbound-method
   const { push } = useRouter();
-  const [isPending, startTransition] = useTransition();
-  const clerkNavRef = useRef<(...args: Parameters<NavigateFunction>) => Promise<void>>();
-  const clerkNavPromiseBuffer = useRef<(() => void)[]>([]);
+  const pathname = usePathname();
+
+  if (typeof window !== 'undefined') {
+    window.__internal_clerk_navFun = to => {
+      if (to === window.location.href.replace(window.location.origin, '')) {
+        push(to);
+        return Promise.resolve();
+      }
 
-  // Set the navigation function reference only once
-  if (!clerkNavRef.current) {
-    clerkNavRef.current = (to, opts) => {
       return new Promise<void>(res => {
-        clerkNavPromiseBuffer.current.push(res);
-        startTransition(() => {
```

Comment:
I believe removing the transition here will re-introduce the original issue, which  is that state updates were happening during the router transition, or that the setActive call was not completing at all because of the navigation triggered by `onBeforeSetActive`.

Is there a way we can retain the use of the transition while keeping the buffer on the window?

Reply:
We'll most likely keep using the transition API - this is removed temporarily so I can cut a snapshot version for testing

Reply:
> setActive call was not completing at all because of the navigation triggered by onBeforeSetActive
Regarding this one, I never managed to reproduce a failed setActive because of us not listening to `isPending`
I *think* the issue was caused by the components being unmounted during navigation but I do agree that keep using transition state is the way to go here